math/mathcore/src/FitUtil.cxx

```diff
@@ -741,7 +741,7 @@ void FitUtil::EvaluateChi2Gradient(const IModelFunction &f, const BinData &data,
             double x1_j = *data.GetCoordComponent(i, j);
             double x2_j = data.GetBinUpEdgeComponent(i, j);
             binVolume *= std::abs(x2_j - x1_j);
-            xc[j] = 0.5 * (x2_j + x1_j);
+            xc[j] = (useBinIntegral) ? x1_j : 0.5 * (x2_j + x1_j);
```

Comment:
So if `useBinIntegral` is true, then `xc` will store the lower bin boundary and not the center. Why is this different, I thought `useBinIntegral` only relates to the "fval", which is now integrated over the bin?

Reply:
Because when using the bin integral one needs the bin boundaries to compute the integral. In this case, the integral of the gradient, which is computed at line 770 below

Reply:
I understand, thanks to the pointer to line 770